src/spikeinterface/sorters/external/mountainsort5.py

```diff
@@ -169,13 +170,21 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
             block_sorting_parameters=scheme2_sorting_parameters, block_duration_sec=p["scheme3_block_duration_sec"]
         )
 
-        scheme = p["scheme"]
-        if scheme == "1":
-            sorting = ms5.sorting_scheme1(recording=recording, sorting_parameters=scheme1_sorting_parameters)
-        elif p["scheme"] == "2":
-            sorting = ms5.sorting_scheme2(recording=recording, sorting_parameters=scheme2_sorting_parameters)
-        elif p["scheme"] == "3":
-            sorting = ms5.sorting_scheme3(recording=recording, sorting_parameters=scheme3_sorting_parameters)
+        assert isinstance(recording, BaseRecording)
+
+        with TemporaryDirectory() as tmpdir:
+            # cache the recording to a temporary directory for efficient reading (so we don't have to re-filter)
+            recording_cached = create_cached_recording(recording=recording, folder=tmpdir)
+
+            scheme = p["scheme"]
+            if scheme == "1":
+                sorting = ms5.sorting_scheme1(recording=recording_cached, sorting_parameters=scheme1_sorting_parameters)
+            elif p["scheme"] == "2":
+                sorting = ms5.sorting_scheme2(recording=recording_cached, sorting_parameters=scheme2_sorting_parameters)
+            elif p["scheme"] == "3":
+                sorting = ms5.sorting_scheme3(recording=recording_cached, sorting_parameters=scheme3_sorting_parameters)
+            else:
+                raise Exception(f"Invalid scheme: {scheme}")
```

Comment:
```suggestion
                raise ValueError(f"Invalid scheme: {scheme} given. scheme must be one of '1', '2' or '3'")
```
What do you think about this? To help guide users to the correct potential schemes.